### ISSUE TITLE
Testbench fails when used on a ComboBox with a newItemHandler

### ISSUE THREAD

**Issue by Wooza** [vaadin/testbench#1009]
When I use TestBench to interact with a ComboBox that uses a NewItemHandler an error occurs.

Sample UI:
```
final VerticalLayout vertLayout = new VerticalLayout();
vertLayout.setSizeFull();
List<String> items = new ArrayList<String>();
for (int i = 0; i < 20; i++) {
	items.add(i + "");
}
ComboBox<String> combo = new ComboBox<>("Test");
combo.setItems(items);
combo.setEmptySelectionAllowed(false);
combo.setNewItemHandler(inputString -> {
	items.add(inputString);
	combo.setItems(items);
	combo.setSelectedItem(inputString);
});
combo.setSelectedItem(items.iterator().next());
vertLayout.addComponent(combo);
setContent(vertLayout);
```

Testbenchcode:
```
@Test
public void changeComboBoxValue() throws Exception {
	setComboBoxValue("abc123");
	setComboBoxValue("11");
	setComboBoxValue("abc123");
}

public void setComboBoxValue(final String value) {
	ComboBoxElement combobox = $(ComboBoxElement.class).first();
	if (combobox.getPopupSuggestions().contains(value)) {
		//Select existing item
		combobox.selectByText(value);
	} else {
		//Enter new item
		combobox.clear();
		combobox.sendKeys(value);
		combobox.sendKeys(Keys.ENTER);
	}
}
```

When using the above code in Testbench with a ChromeDriver the interaction with the browser stops after selecting the value "11". A short while after that I get the following error message a few times:

> Mär 02, 2018 1:54:47 PM org.apache.http.impl.execchain.RetryExec execute
INFORMATION: I/O exception (java.net.BindException) caught when processing request to {}->http://localhost:48913: Address already in use: connect
Mär 02, 2018 1:54:47 PM org.apache.http.impl.execchain.RetryExec execute
INFORMATION: Retrying request to {}->http://localhost:48913

After that the test fails with 

> org.openqa.selenium.NoSuchElementException: Vaadin could not find elements with the selector (//com.vaadin.ui.ComboBox)[0]

Tested with Vaadin version 8.3.1 and the respective testbench-api. This seems to happen with all major browsers (IE, Chrome, FF), although only Chrome prints the error messages whereas IE and FF did not respond in anyway.

**Comment by focbenz** [comment]
The same issue could be observed on our buildserver since about a week failing most of the UI-Tests:
`error	22-Feb-2018 08:24:16	Feb 22, 2018 8:24:16 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:16	INFORMATION: I/O exception (java.net.BindException) caught when processing request to {}->http://localhost:12881: Address already in use: connect
error	22-Feb-2018 08:24:16	Feb 22, 2018 8:24:16 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:16	INFORMATION: Retrying request to {}->http://localhost:12881
error	22-Feb-2018 08:24:16	Feb 22, 2018 8:24:16 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:16	INFORMATION: I/O exception (java.net.BindException) caught when processing request to {}->http://localhost:12881: Address already in use: connect
error	22-Feb-2018 08:24:16	Feb 22, 2018 8:24:16 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:16	INFORMATION: Retrying request to {}->http://localhost:12881
error	22-Feb-2018 08:24:26	Feb 22, 2018 8:24:26 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:26	INFORMATION: I/O exception (java.net.BindException) caught when processing request to {}->http://localhost:12881: Address already in use: connect
error	22-Feb-2018 08:24:26	Feb 22, 2018 8:24:26 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:26	INFORMATION: Retrying request to {}->http://localhost:12881
error	22-Feb-2018 08:24:26	Feb 22, 2018 8:24:26 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:26	INFORMATION: I/O exception (java.net.BindException) caught when processing request to {}->http://localhost:12881: Address already in use: connect
error	22-Feb-2018 08:24:26	Feb 22, 2018 8:24:26 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:26	INFORMATION: Retrying request to {}->http://localhost:12881
error	22-Feb-2018 08:24:26	Feb 22, 2018 8:24:26 AM org.apache.http.impl.execchain.RetryExec execute
error	22-Feb-2018 08:24:26	INFORMATION: I/O exception (java.net.BindException) caught when processing request to {}->http://localhost:12881: Address already in use: connect`

**Comment by johannesh2** [comment]
Is this same exception about coming from other tests also and not related to the ComboBox or did I understand it wrong

**Comment by focbenz** [comment]
These messages result from the remaining tests not being able to connect through the webdriver to the browser anymore after trying to set a value in one of these ComboBoxes. So in this case all further Vaadin Testbench methods in the same test class fail with the same error.

**Comment by Artur-** [comment]
Seems like a regression in 8.3.1 and 8.3.0 works ok

**Comment by Artur-** [comment]
Test project: https://github.com/Artur-/tbcombo (does not run IT automatically)
Likely cause: https://github.com/vaadin/framework/issues/10284
Reason: `VComboBox.isWorkPending()` returns true even though `ComboBox` is no longer busy. TestBench waits until it times out for `isWorkPending()` to return false

**Comment by focbenz** [comment]
With tsuoanttila's referenced commit I guess this will only be available starting with 8.4.0.alpha1 ?
